### PR TITLE
Feature - Access database in tests

### LINKED ISSUE #83: Accessing the database in api-tests
**Describe the new feature you would like to see**
Some of the endpoints that are tested in the api-tests update the database. It would be beneficial to check that the expected updates to the database is done correctly. 

**Describe the solution you'd like**
An easy method for accessing the same database as used by the test client should be found and setup this for the appropriate test.

**How will this feature affect the current Threat Model?**
No changes

### PR DESCRIPTION
**Pull request by GodVenn**
Closes #83 
~~This PR includes the commits from #101 , the plan is to rebase once that one is merged
Only the last 3 commits belong to this PR~~

### CODE REVIEW

**Review by vetlek — COMMENTED** [review]
Hard to review this PR (see what changes are relevant for this PR) when it is based on another PR that have not yet been merged into main

File: backend/tests/test_crud.py [hunk 1]
```diff
@@ -1,9 +1,142 @@
...
+from flotilla.database.models import (
+    EventDBModel,
+    ReportDBModel,
+    ReportStatus,
+    RobotDBModel,
+)
 
-def test_get_robots(session):
+good_ids: list[int] = [1, 2]
+bad_ids: list[int] = [-1, 13, 56]
```
Comment (anetteu) [inline backend/tests/test_crud.py:27]: Should good/bad ids be valid/unvalid ids?

File: backend/tests/test_crud.py [hunk 2]
```diff
@@ -1,9 +1,142 @@
...
+    bad_ids,
+)
+def test_read_event_by_id_throws_404(event_id: int, session):
+    with pytest.raises(HTTPException) as e:
+        event: EventDBModel = read_event_by_id(db=session, event_id=event_id)
+        assert e.value.status_code == HTTPStatus.NOT_FOUND.value
+
+
+def test_create_report(session):
+    robot_id = 1
```
Comment (anetteu) [inline backend/tests/test_crud.py:103]: typehint?

**Review by anetteu — APPROVED** [review]
Looks good! Please address minor comments before merging

**Pull request merged**